### PR TITLE
refactor: simplify sbb-tag-group logic

### PR DESCRIPTION
**Pull request by kyubisation**
Previously the current solution was not possible in Stencil.js. It is now however possible with Lit.

### CODE REVIEW

File: src/components/tag/tag/tag.ts [hunk 1]
```diff
@@ -134,6 +95,17 @@ export class SbbTagElement extends SbbIconNameMixin(
     this._didChange.emit();
   }
 
+  protected override willUpdate(changedProperties: PropertyValues<this>): void {
+    super.willUpdate(changedProperties);
+    const tagGroup = this.closest('sbb-tag-group');
+    if (tagGroup && !tagGroup.multiple && changedProperties.has('checked') && this.checked) {
+      for (const tag of tagGroup?.tags.filter((t) => t !== this) ?? []) {
+        tag.checked = false;
+        tag.requestUpdate('checked');
```
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/tag/tag/tag.ts:104]: Yes, but the attribute is only updated when called with requestUpdate('checked'). Although I could refactor `checked` to correlate to the attribute, as Jeri and I did with value and name.

[the remark this answers is not in the document]
Comment (TomMenga) [reply src/components/tag/tag/tag.ts:104]: Mmm, strange. I expected Lit to call `requestUpdate` when `checked` is set. 
Just out of curiosity, I'll investigate this further.
In the meantime, it's fine as it is :)

**Review by TomMenga — APPROVED** [review]
LGTM 👍

File: src/components/tag/tag/tag.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/tag/tag/tag.ts:104]: Actually, looking at the code, it should update.
Hm... I'll play around with it a bit more.

[the remark this answers is not in the document]
Comment (kyubisation) [reply src/components/tag/tag/tag.ts]: I removed it and ran tests again and it seems to still work. Strange. I thought it was breaking tests...
But it's certainly better this way. Thanks for pointing it out. 👍

**Pull request merged**